### PR TITLE
Move some old security controls to KubeletFlags and mark them deprecated

### PR DESCRIPTION
**Pull request by mtaufen**
Related to @timstclair's comment: https://github.com/kubernetes/kubernetes/pull/53833#issuecomment-355399163

```release-note
NONE
```

### CODE REVIEW

**Comment by tallclair** [comment]
/lgtm
(need to update bazel)

**Comment by mtaufen** [comment]
Ah, looks like the test I wrote to catch people removing fields caught me removing a field :D

**Comment by mtaufen** [comment]
re-adding label after minor fixup

**Comment by liggitt** [comment]
> the test I wrote to catch people removing fields caught me removing a field 

Those are the best. /waves to past self

**Comment by mtaufen** [comment]
/waves with gratitude to past @liggitt as well :D

File: cmd/kubelet/app/options/options.go [hunk 1]
```diff
@@ -366,6 +383,14 @@ func (f *KubeletFlags) AddFlags(fs *pflag.FlagSet) {
 	// TODO(#54161:v1.11.0): Remove --enable-custom-metrics flag, it is deprecated.
 	fs.BoolVar(&f.EnableCustomMetrics, "enable-custom-metrics", f.EnableCustomMetrics, "Support for gathering custom metrics.")
 	fs.MarkDeprecated("enable-custom-metrics", "will be removed in a future version")
+	fs.BoolVar(&f.AllowPrivileged, "allow-privileged", f.AllowPrivileged, "If true, allow containers to request privileged mode.")
```
Comment (liggitt) [inline cmd/kubelet/app/options/options.go]: Add comments tracking which release these can be removed in

Comment (mtaufen) [reply cmd/kubelet/app/options/options.go]: done

**Comment by mtaufen** [comment]
ah, some other PRs of mine have merged and this needs an update now

**Comment by mtaufen** [comment]
re-adding label, since only change since last lgtm was a comment tracking field deprecation.

**Pull request merged**